### PR TITLE
Clearing up hook naming changes.

### PR DESCRIPTION
**Pull request by knjackson30974**
**Jira:** https://jira.duraspace.org/browse/ISLANDORA-1637
# What does this Pull Request do?

Addresses hook naming changes in xml_form_builder. 
# How should this be tested?

Check that hook names match hook names in xml_form_builder.api.php
# Background context:
# Additional Notes:
- **Does this change require documentation to be updated?** No.
- **Does this change add any new dependencies?** No.
- **Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)?** No.
- **Could this change impact execution of existing code?** If any implementations of the hooks were missed perhaps. Would be minor. 

**Tagging:** @DiegoPino 

---

Kylea Jackson

_Developer_

**[discoverygarden inc.](http://www.discoverygarden.ca) | Managing Digital Content**

### CODE REVIEW

File: tests/islandora_solution_pack_test/islandora_test_cm.module
Comment (jordandukart) [inline tests/islandora_solution_pack_test/islandora_test_cm.module]: Tests probably doesn't need a deprecated could likely just remove it

**Comment by jordandukart** [comment]
This has undergone review internally and would be nice to have another external committer verify this so it can move forward.

Thanks!

**Comment by whikloj** [comment]
This and all the assorted solution pack ones have to go in together right?

**Comment by adam-vessey** [comment]
@whikloj Not strictly, no. I believe the other referenced pulls depend on this one, but this one could be merged without the individual implementations since we still invoke the old hook names (for now and at least until after the next release; they're being deprecated).

**Comment by whikloj** [comment]
@adam-vessey cool thanks. I will try to look at this and give it a test today or tomorrow.

**Comment by DiegoPino** [comment]
@adam-vessey, @knjackson30974, @ruebot @whikloj , should we proceed with this? This is marked as bug and fix version is 1.6, so i suppose we will have a bunch of release pulls also?
Lastly: https://github.com/Islandora/islandora/wiki/Creating-and-Working-With-XML-Forms#adding-a-form-to-the-list-of-forms this needs some love if we merge this.

**Comment by qadan** [comment]
@ruebot While this is listed as a 'bug', it's basically just because I don't appear to be able to switch https://jira.duraspace.org/browse/ISLANDORA-1637 over to a 'Code Task'? Dunno what JIRA's problem is, but anyway, since this is a developer-focused 'codebase improvement' change that is essentially invisible to end users, it's better classified as a code task and shouldn't need a release pull.

**Comment by jordandukart** [comment]
This has been sitting since March and wasn't included in the last release, @dannylamb, @Islandora/7-x-1-x-committers this needs some eyes on it.

**Comment by dannylamb** [comment]
Bumping @Islandora/7-x-1-x-committers on this

**Comment by jordandukart** [comment]
This and all of its other PRs referenced within here and https://jira.duraspace.org/browse/ISLANDORA-1637 has been sitting stale since March. @dannylamb, @Islandora/7-x-1-x-committers.

**Comment by dannylamb** [comment]
Taking the reigns on this one.

**Comment by DiegoPino** [comment]
@dannylamb citing myself from May

> @adam-vessey, @knjackson30974, @ruebot @whikloj , should we proceed with this? This is marked as bug and fix version is 1.6, so i suppose we will have a bunch of release pulls also?
> Lastly: https://github.com/Islandora/islandora/wiki/Creating-and-Working-With-XML-Forms#adding-a-form-to-the-list-of-forms this needs some love if we merge this.

Bug was moved to improvement but no fix version.
 @knjackson30974 i can put 1.8 there if you want.

I think probably the Wiki issue is the more concerning, since the naming affects documentation. And we need to take care of all other other pulls right? 
Lastly, since these hooks could be used potentially in a lot of custom modules this should go to the list¿? Not needed?

**Comment by dannylamb** [comment]
@DiegoPino Sounds like a documentation task for the release.  I'm looking into all the other pulls now.

**Comment by dannylamb** [comment]
I've reviewed all the associated PRs.  Pulled all them down and tested on a fresh vagrant, and nothing seems out of place.

There are tests that are failing for book and newspaper that I'm looking into now before pulling the trigger.  They appear to still be failing even when run locally.

There may be others out there with hooks they need to update, so I'll post to the list about it, too,

**Comment by dannylamb** [comment]
Tests in Collection and Newspaper solutions packs are failing against their respective pulls.  I've re-run them against head and it's all green.  So I can't bring these in until we get that sorted out.

If we can get to the bottom of it quickly, I'll still consider pulls against release branches if someone wants to go through the hassle of making all of 'em.

**Comment by DiegoPino** [comment]
@dannylamb i will give this a look later today, thanks

**Comment by DonRichards** [comment]
@DiegoPino were you able to test this? This is a blocker for a PR in scholar so I'm following up on it.

**Comment by DiegoPino** [comment]
@DonRichards sorry, explain the blocker thing?

**Comment by DiegoPino** [comment]
@knjackson30974 @jordandukart i will make sure this goes into the 7.x-1.9 release. There are a lot of small pulls to test (all look fine i know), but i have to make sure all works and we have no other changes missing. Will have this done by Thursday Febr 9 2017.

**Comment by DonRichards** [comment]
@DiegoPino @whikloj made a comment within https://github.com/Islandora/islandora_scholar/pull/227 that this was a blocker for 227 to be merged. 227 addresses hook naming changes in xml_form_builder

**Comment by DiegoPino** [comment]
@DonRichard, this is a task that needs to be done, and yes, all hooks need to be named in a certain way on implementing modules( a long list of referenced pulls at the beginning of this) and all that depends on this one (which is the one that invokes the hooks) but it is not a "blocker".  I will have this chain of pulls tested and merged by Feb. 9th so no need to worry.

**Comment by whikloj** [comment]
@DonRichards This is not a "blocker". Those other PRs depend on this one.

This PR makes changes to the xml_forms hook naming, those other PRs change the various modules to change their use of the hooks to use the new names (new as in the ones added in this PR).

So you can't merge the other PRs until you merge this one.

But they are not a blocker.

**Review by DiegoPino — APPROVED** [review]
Tested and working(standalone) without any changes to the other modules. Nothing breaks. Will merge now and test all the other depending ones. Just a side note to myself, fix as a code task https://github.com/Islandora/islandora_xml_forms/pull/221/files#diff-b893f498b9bca6b1453c419abd533b59R74 small little type. @knjackson30974 thanks a lot for this one.

**Comment by DiegoPino** [comment]
After merge note to myself: 
@whikloj offered to make pull request for Islandora solution pack disk image, which is the one that was missing. Thanks @whikloj!

**Comment by jordandukart** [comment]
Thanks for merging @DiegoPino!

**Pull request merged**
